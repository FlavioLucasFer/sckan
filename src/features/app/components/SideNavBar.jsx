import React from 'react';
import { useHistory } from 'react-router-dom';
import { SideNav, SideNavItem, Icon } from 'react-materialize';

import { 
	COMPANY_ROUTE, 
	LOGIN_ROUTE, 
	PROJECT_ROUTE, 
	SPRINT_ROUTE, 
	TASK_ROUTE, 
	USER_ROUTE,
} from 'core/utils/routes';

import { AppNameDiv, AppNameH1 } from './StyledComponents';

function SideNavBar() {
	const { push } = useHistory();

	return (
		<SideNav id='side-navbar'
			data-testid='side-navbar'>
			<AppNameDiv>
				<AppNameH1>
					Sckan
				</AppNameH1>
			</AppNameDiv>
			
			<SideNavItem waves
				data-testid='users-sidenav-item'
				href={USER_ROUTE}
				icon={<Icon small>manage_accounts</Icon>}
				onClick={() => push(USER_ROUTE)}>
				Users
			</SideNavItem>

			<SideNavItem waves
				data-testid='companies-sidenav-item'
				href={COMPANY_ROUTE}
				icon={<Icon small>business</Icon>}
				onClick={() => push(COMPANY_ROUTE)}>
				Companies
			</SideNavItem>

			<SideNavItem waves
				data-testid='tasks-sidenav-item'
				href={TASK_ROUTE}
				icon={<Icon small>task</Icon>}
				onClick={() => push(TASK_ROUTE)}>
				Tasks
			</SideNavItem>

			<SideNavItem waves
				data-testid='sprints-sidenav-item'
				href={SPRINT_ROUTE}
				icon={<Icon small>wysiwyg</Icon>}
				onClick={() => push(SPRINT_ROUTE)}>
				Sprints
			</SideNavItem>

			<SideNavItem waves
				data-testid='projects-sidenav-item'
				href={PROJECT_ROUTE}
				icon={<Icon small>source</Icon>}
				onClick={() => push(PROJECT_ROUTE)}>
				Projects
			</SideNavItem>

			<SideNavItem divider />

			<SideNavItem waves
				data-testid='logout-sidenav-item'
				href={LOGIN_ROUTE}
				icon={<Icon small>logout</Icon>}
				onClick={() => push(LOGIN_ROUTE)}>
				Log-out
			</SideNavItem>
		</SideNav>
	);
}

export default SideNavBar;