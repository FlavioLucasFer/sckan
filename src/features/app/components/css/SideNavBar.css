.sidenav-li {
	height: 50px;
	width: 100%;
	font-size: medium;
	color: #000;
	text-align: left;
	text-transform: capitalize;
	padding-right: 0;
	margin: 2.5px 0px;
	border-radius: 0 !important;
}

.sidenav-li:hover {
	background-color: rgba(0, 0, 0, .1);
}

.sidenav-span {
	margin-left: 20px;
}

.li-icon {
	position: absolute;
	top: 0;
	right: 0;
	
	font-size: 100px !important;
	z-index: -1;
}

.modal-trigger {
	padding: 0;
}