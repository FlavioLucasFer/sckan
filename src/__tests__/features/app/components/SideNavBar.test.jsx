import React from 'react';
import { shallow } from 'enzyme';

import SideNavBar from 'features/app/components/SideNavBar';
import { 
	COMPANY_ROUTE, 
	HOME_ROUTE, 
	LOGIN_ROUTE, 
	PROJECT_ROUTE, 
	SPRINT_ROUTE, 
	TASK_ROUTE, 
	USER_ROUTE 
} from 'core/utils/routes';

const mockHistory = {
	push: jest.fn(),
};

const mockLocation = {
	pathname: 'localhost:3000',
};

jest.mock('react-router-dom', () => ({
	...(jest.requireActual('react-router-dom')),
	useHistory: () => (mockHistory),
	useLocation: () => (mockLocation),
}));

describe('Testing SideNavBar app feature component', () => {
	function testNavigation(route) {
		const wrapper = shallow(
			<SideNavBar />
		);

		wrapper.find({ route })
			.dive()
			.simulate('click');

		expect(mockHistory.push).toBeCalledWith(route);
	}
	
	afterAll(() => {
		jest.resetAllMocks();
	});
	
	it('should render correctly', () => {
		const wraaper = shallow(
			<SideNavBar />
		);

		expect(wraaper.debug()).toMatchSnapshot();
	});

	it('should navigate to home screen when click on home SideNavItem', () => {
		testNavigation(HOME_ROUTE);
	});

	it('should navigate to users list screen when click on users SideNavBar', () => {
		testNavigation(USER_ROUTE);
	});

	it('should navigate to companies list screen when click on companies SideNavBar', () => {
		testNavigation(COMPANY_ROUTE);
	});

	it('should navigate to tasks screen when click on tasks SideNavBar', () => {
		testNavigation(TASK_ROUTE);
	});

	it('should navigate to sprints list screen when click on sprints SideNavBar', () => {
		testNavigation(SPRINT_ROUTE);
	});

	it('should navigate to projects list screen when click on projects SideNavBar', () => {
		testNavigation(PROJECT_ROUTE);
	});

	it('should navigate to login screen when click on logout SideNavBar and confirm logout modal', () => {
		const wrapper = shallow(
			<SideNavBar />
		);

		wrapper.find({ id: 'logout-modal' })
			.dive()
			.find({ id: 'logout-modal' })
			.dive()
			.find({ id: 'logout-modal' })
			.dive()
			.find({ title: 'Yes' })
			.simulate('click');

		expect(mockHistory.push).toBeCalledWith(LOGIN_ROUTE);
	});
});