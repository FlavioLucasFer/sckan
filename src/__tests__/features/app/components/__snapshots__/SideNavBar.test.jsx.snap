// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SideNavBar app feature component should render correctly 1`] = `
"<ul id=\\"sidenav\\" data-testid=\\"sidenav\\">
  <SideNavItem route=\\"/\\" label=\\"Home\\" icon=\\"home\\" data-testid=\\"home-sidenav-item\\" first={true} />
  <SideNavItem route=\\"/users\\" label=\\"Users\\" icon=\\"manage_accounts\\" data-testid=\\"users-sidenav-item\\" />
  <SideNavItem route=\\"/companies\\" label=\\"Companies\\" icon=\\"business\\" data-testid=\\"companies-sidenav-item\\" />
  <SideNavItem route=\\"/tasks\\" label=\\"Tasks\\" icon=\\"task\\" data-testid=\\"tasks-sidenav-item\\" />
  <SideNavItem route=\\"/sprints\\" label=\\"Sprints\\" icon=\\"wysiwyg\\" data-testid=\\"sprints-sidenav-item\\" />
  <SideNavItem route=\\"/projects\\" label=\\"Projects\\" icon=\\"source\\" data-testid=\\"projects-sidenav-item\\" />
  <Divider />
  <a className=\\"modal-trigger\\" href=\\"#logout-modal\\">
    <SideNavItem label=\\"Log-out\\" icon=\\"logout\\" data-testid=\\"logout-sidenav-item\\" />
  </a>
  <LogOutModal id=\\"logout-modal\\" />
</ul>"
`;
